### ISSUE TITLE
Customising the system wide default configuration

### ISSUE THREAD

**Issue by ysalmon** [pyzo/pyzo#511]
Hello,
we have been using Pyzo for some years at my school and I am preparing the deployment of the latest version. I pushed and unpacked the latest (4.4.2) version on a test machine but when running pyzo, the welcome window is briefly displayed before the application crashes. The console tells us that
```
yann.salmon@c204btest:~$ pyzo
Started our command server
loading qt French: failed
loading pyzo French: ok
Uncaught Python exception: 'Dict' object has no attribute 'allowFloatingShell'
  File "/home/almar/miniconda3/lib/python3.5/site-packages/cx_Freeze/initscripts/__startup__.py", line 14, in run
  File "/home/almar/miniconda3/lib/python3.5/site-packages/cx_Freeze/initscripts/Console.py", line 26, in run
  File "/home/almar/dev/pyzo/pyzo/__main__.py", line 56, in <module>
  File "/home/almar/dev/pyzo/pyzo/__main__.py", line 52, in main
  File "/opt/pyzo/source/pyzo/__init__.py", line 264, in start
    MainWindow(None, appLocale)
  File "/opt/pyzo/source/pyzo/core/main.py", line 96, in __init__
    self._populate()
  File "/opt/pyzo/source/pyzo/core/main.py", line 170, in _populate
    if pyzo.config.settings.allowFloatingShell:
  File "/opt/pyzo/source/pyzo/util/zon.py", line 65, in __getattribute__
    return object.__getattribute__(self, key)
```

Pyzo 4.4.2 runs fine on my own computer so I am wondering.

Is this happening because I am also pushing a custom ```defaultConfig.ssdf``` that was created from the default file of an earlier version of Pyzo (my aim in doing so is to preselect French, the path to our python interpreter and set the font size to something bigger) ?

**Comment by almarklein** [comment]
> Is this happening because I am also pushing a custom defaultConfig.ssdf

Yes. I would not recommend using a different version of the default config. When a new config setting gets added to Pyzo, it also gets added to the default config, and pushing your own version will then cause the config to have missing values ...

**Comment by ysalmon** [comment]
This makes sense.

Overwriting the file is not a very good way to customise the default config, but due to a bug in fabric https://github.com/fabric/fabric/issues/1502 I cannot just sed the file.

If I just append my customisations at the end of the file, will they take precedence ?

**Comment by almarklein** [comment]
> If I just append my customisations at the end of the file, will they take precedence ?

mmm, I'm not sure, but would not be surprised if it would. Try! :)

**Comment by ysalmon** [comment]
Will do when I have access to relevant computer. 
Is there a better way to customise the systemwide configuration ? An API or CLI or maybe a separate file or directory that would be read à la Emacs site-start.d ?

**Comment by almarklein** [comment]
Not at the moment.

**Comment by ysalmon** [comment]
I believe this will do the trick https://github.com/pyzo/pyzo/pull/512

**Comment by almarklein** [comment]
Closed by #512
